File: Builder/PHP/ExcelReportBuilder.php
<?php

require_once "IReportBuilder.php";
require_once "Report.php";

class ExcelReportBuilder implements IReportBuilder {
    private Report $report;

    public function buildContent()
    {
        if (!isset($this->report)) {
            $this->report = new Report;
        }

        $this->report->setContent("Tables");
    }

    public function buildFile()
    {
        if (!isset($this->report)) {
            $this->report = new Report;
        }

        $this->report->setFilename("Report.xlsx");
    }

    public function getResult() {
        return $this->report;
    }
}

File: Builder/Python 3/ExcelReportBuilder.py
from IReportBuilder import IReportBuilder
from Report import Report

class ExcelReportBuilder(IReportBuilder):
  def buildContent(self):
    if not self._report:
      self._report = Report()
    
    self._report.content = "Tables"

  def buildFile(self):
    if not self._report:
      self._report = Report()
    
    self._report.filename = "Report.xlsx"

  def getResult(self):
    return self._report